code_generation/quantity.py

```diff
@@ -12,12 +12,24 @@ def __init__(self, name):
         self.defined_for_scopes = []
         log.debug("Setting up new Quantity {}".format(self.name))
 
-    # the scopes, in which a quantity is used as an output is tracked in the output_scopes list.
-    # This check is triggered for every producer.
-    # If a quantity is already used within a given scope as output, this will result in an exception.
+    def __str__(self) -> str:
+        return self.name
+
+    def __repr__(self) -> str:
+        return self.name
+
     def reserve_scope(self, scope):
+        """
+        Function to reserve a scope for a given quantity. The scopes, in which a quantity is used
+        as an output are tracked in the output_scopes list.
+        If a quantity is already used within a given scope as output, this will result in an exception.
+        This check is triggered for every Producer.
+        """
         log.debug("Checking {} / scope {}".format(self.name, scope))
-        if scope not in self.defined_for_scopes:
+        if (scope == "global" and self.defined_for_scopes == []) or (
+            scope not in self.defined_for_scopes
+            and "global" not in self.defined_for_scopes
```

Comment:
this line must be scope != "global". Otherwise global could be added via this part of the or even if defined_for_scopes is not empty and the first part fails.

Reply:
changed